train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -1,12 +1,28 @@
 package hu.bme.mit.train.controller;
 
+import java.util.Timer;
+import java.util.TimerTask;
+
 import hu.bme.mit.train.interfaces.TrainController;
 
 public class TrainControllerImpl implements TrainController {
 
 	private int step = 0;
 	private int referenceSpeed = 0;
 	private int speedLimit = 0;
+	private Timer timer;
+	
+	public TrainControllerImpl()
+	{
+		timer = new Timer();
+		timer.schedule(new TimerTask() {
+
+            @Override
+            public void run() {
+                followSpeed();
+            }
+        }, 0, 1000);
```

Comment:
Is it a correct time period? Reaction time is too big! if the train goes with 100 km/h = 360 m/s it went 360 m till it changes the speed